### PR TITLE
UPSTREAM: 00000: make delete waits match on UID

### PR DESCRIPTION
**Pull request by deads2k**
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15509/consoleText indicates that UIDs don't match so the wait never finishes.

@liggitt @soltysh @juanvallejo

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -252,7 +254,23 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
 		}
 		options.PropagationPolicy = &policy
 
-		return o.deleteResource(info, options)
+		response, err := o.deleteResource(info, options)
+		if err != nil {
+			return err
+		}
+		if status, ok := response.(*metav1.Status); ok && status.Details != nil {
+			uidMap[kubectlwait.ResourceLocation{Namespace: info.Namespace, Name: info.Name}] = status.Details.UID
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: does delete never handle multiple resource types? I expected at least group/resource in the key as well

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go [hunk 2]
```diff
@@ -294,23 +313,25 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
...
 		}
 	}
 
-	if err := resource.NewHelper(info.Client, info.Mapping).DeleteWithOptions(info.Namespace, info.Name, deleteOptions); err != nil {
-		return cmdutil.AddSourceToErr("deleting", info.Source, err)
+	deleteResponse, err := resource.NewHelper(info.Client, info.Mapping).DeleteWithOptions(info.Namespace, info.Name, deleteOptions)
+	if err != nil {
+		return nil, cmdutil.AddSourceToErr("deleting", info.Source, err)
 	}
+	fmt.Printf("#### %v\n", spew.Sdump(deleteResponse))
```
Comment (liggitt) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: drop debug info

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go (hunk 1 above)
Comment (deads2k) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: > does delete never handle multiple resource types? I expected at least group/resource in the key as well

oops.  Yep.

Comment (deads2k) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: > does delete never handle multiple resource types? I expected at least group/resource in the key as well

fixed and test updated

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go (hunk 2 above)
Comment (deads2k) [reply vendor/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go]: > drop debug info

fixed

**Comment by deads2k** [comment]
- [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15544/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15545/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15546/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15547/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15548/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15549/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15550/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15551/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15552/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15553/
 - [ ] https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_cmd/15554/

**Comment by deads2k** [comment]
upstream here https://github.com/kubernetes/kubernetes/pull/66136

Will rename after green.

**Comment by deads2k** [comment]
renamed commit

**Comment by deads2k** [comment]
merging

**Pull request merged**